File: src/main/java/tg/Ipnet/efunerailles/Service/DefuntService.java
package tg.Ipnet.efunerailles.Service;


import java.util.List;
import java.util.Optional;

import tg.Ipnet.efunerailles.Entity.Defunt;

public interface DefuntService {

	List<Defunt> getAllDefunts();
    Optional<Defunt> getDefuntById(Long id);
    Defunt createDefunt(Defunt defunt);
    Defunt updateDefunt(Long id, Defunt defuntDetails);
    void deleteDefunt(Long id);
}

File: frontend/src/app/core/services/defunt.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Defunt } from '../models/defunt.model';

@Injectable({
  providedIn: 'root'
})
export class DefuntService {
  private apiUrl = 'http://localhost:8181/api/defunts'; // À confirmer

  constructor(private http: HttpClient) { }

  getAllDefunts(): Observable<Defunt[]> {
    return this.http.get<Defunt[]>(this.apiUrl);
  }

  getDefuntById(id: number): Observable<Defunt> {
    return this.http.get<Defunt>(`${this.apiUrl}/${id}`);
  }

  // Dans defunt.service.ts
createDefunt(defunt: Partial<Defunt>): Observable<Defunt> {
  return this.http.post<Defunt>(this.apiUrl, defunt);
}

  updateDefunt(id: number, defunt: Defunt): Observable<Defunt> {
    return this.http.put<Defunt>(`${this.apiUrl}/${id}`, defunt);
  }

  deleteDefunt(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
